### PR TITLE
[1LP][RFR]Initial Prototype of SUI Console Tests

### PR DESCRIPTION
**Pull request by kedark3**
Purpose or Intent
=================
__Updating tests__for SSUI Console, to make it complete. In previous PRs the test was added just as a stub, and was not asserting anything. Now it should test working of console. 

{{pytest: -v --long-running cfme/tests/ssui/test_ssui_myservice.py -k 'test_vm_console' --use-provider vsphere6-nested --use-provider rhv41}}

### CODE REVIEW

File: cfme/tests/ssui/test_ssui_myservice.py [hunk 1]
```diff
@@ -52,15 +65,70 @@ def test_myservice_crud(appliance, setup_provider, context, order_catalog_item_i
...
+                func_kwargs={'ensure_user': True},
+                fail_condition=lambda result: result.rc != 0, delay=1, num_sec=10)
+            # if file was created in previous steps it will be removed here
+            # we will get instance of SSHResult
+            # Sometimes Openstack drops characters from word 'blather' hence try to remove
+            # file using partial file name. Known issue, being worked on.
+            command_result = vm_ssh_client.run_command("rm blather", ensure_user=True)
+            assert command_result
+
+        except:
```
Comment (rlbabyuk) [inline cfme/tests/ssui/test_ssui_myservice.py]: Required: One small thing here: specify an exception type here (as an option Exception can be used here)

File: cfme/tests/ssui/test_ssui_myservice.py [hunk 2]
```diff
@@ -52,15 +65,70 @@ def test_myservice_crud(appliance, setup_provider, context, order_catalog_item_i
         my_service.delete()
 
 
-@pytest.mark.uncollectif(lambda: current_version() < '5.8')
+@pytest.mark.uncollectif(lambda: current_version() < '5.8' or current_version() > '5.9')
```
Comment (rlbabyuk) [inline cfme/tests/ssui/test_ssui_myservice.py]: **Question**: What versions should this test run for? 5.8, 5.9 and ignore upstream?

Comment (kedark3) [reply cfme/tests/ssui/test_ssui_myservice.py]: Hey @rlbabyuk  it should run for 5.8 only, support for 5.7 will be added in next sprints and then 5.9.

File: cfme/tests/ssui/test_ssui_myservice.py [hunk 3]
```diff
@@ -52,15 +65,70 @@ def test_myservice_crud(appliance, setup_provider, context, order_catalog_item_i
...
 @pytest.mark.parametrize('context', [ViaSSUI])
 @pytest.mark.parametrize('order_catalog_item_in_ops_ui', [['console_test']], indirect=True)
-def test_vm_console(appliance, setup_provider, context, configure_websocket,
-        order_catalog_item_in_ops_ui):
+def test_vm_console(request, appliance, setup_provider, context, configure_websocket,
+        order_catalog_item_in_ops_ui, take_screenshot, console_template):
     """Test Myservice VM Console in SSUI."""
     catalog_item = order_catalog_item_in_ops_ui
     service_name = catalog_item.name
+    console_vm_username = credentials[catalog_item.provider.data.templates.get('console_template')
```
Comment (rlbabyuk) [inline cfme/tests/ssui/test_ssui_myservice.py]: **Required**: console_vm_username = credentials[catalog_item.provider.data.templates.console_template.creds].username
As we can access conf values as attr
Please do the same for password

File: cfme/tests/ssui/test_ssui_myservice.py [hunk 4]
```diff
@@ -52,15 +65,70 @@ def test_myservice_crud(appliance, setup_provider, context, order_catalog_item_i
...
+            # Enter Username:
+            vm_console.send_keys("{}".format(console_vm_username))
+            assert vm_console.wait_for_text(text_to_find="Password", timeout=200), ("VM Console"
+            " didn't prompt for Password")
+            # Enter Password:
+            vm_console.send_keys("{}".format(console_vm_password))
+            logger.info("Wait to get the '$' prompt")
+            if catalog_item.provider.one_of(VMwareProvider):
+                vm_console.wait_for_text(text_to_find=catalog_item.provider.data
+                    .templates.get('console_template')['prompt_text'], timeout=200)
```
Comment (rlbabyuk) [inline cfme/tests/ssui/test_ssui_myservice.py]: see comment https://github.com/ManageIQ/integration_tests/pull/5655/files#r146864996

File: cfme/tests/ssui/test_ssui_myservice.py [hunk 5]
```diff
@@ -52,15 +65,70 @@ def test_myservice_crud(appliance, setup_provider, context, order_catalog_item_i
...
+                time.sleep(15)
+            user_count_after_login = vm_ssh_client.run_command("who --count", ensure_user=True)
+            # Number of users before login would be 0 and after login would be 180
+            # If below assertion would fail user_count_after_login is also 0, denoting login failed
+            assert (user_count_before_login.output.split('=')[-1].strip() <
+                    user_count_after_login.output.split('=')[-1].strip()), "Login Failed"
+            logger.info("Wait to get the '$' prompt")
+
+            vm_console.wait_for_text(text_to_find=catalog_item.provider.data.templates.
+                get('console_template')['prompt_text'], timeout=200)
```
Comment (rlbabyuk) [inline cfme/tests/ssui/test_ssui_myservice.py]: here too: [comment](https://github.com/ManageIQ/integration_tests/pull/5655/files#r146864996)

File: cfme/tests/ssui/test_ssui_myservice.py (hunk 2 above)
Comment (rlbabyuk) [reply cfme/tests/ssui/test_ssui_myservice.py]: @kedark3 please update current_version() >= '5.9', if 5.9 should be skipped

Comment (kedark3) [reply cfme/tests/ssui/test_ssui_myservice.py]: But then it will strictly just do == '5.9' right? shouldn't it be >= 5.9?

**Comment by dajoRH** [comment]
Lint report for commit bc7183161825538e67ed04c2c77d0624af20dd1c:

`cfme/tests/ssui/test_ssui_myservice.py`:
- :bangbang: Line 91:101: **E501** *line too long (101 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3bcf531d1252547a44101b6edaa5a27c17e4f457:

`cfme/tests/ssui/test_ssui_myservice.py`:
- :red_circle: Line 4:1: **F401** *'time' imported but unused*
- :red_circle: Line 7:1: **F401** *'cfme.infrastructure.provider.virtualcenter.VMwareProvider' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by rlbabyuk** [comment]
@kedark3 please also check PRT failures

**Comment by kedark3** [comment]
@rlbabyuk Thanks for approval. Did you not move it to 1LP because of PRT Failures? I have kicked another PRT run, I believe their should not be failure, since it works correctly on my personal environment, I am watching PRT run myself to figure out what's wrong.

**Comment by kedark3** [comment]
for future reviewer, in  /trackerbot/pr/run/17321 58z passed with RHV41, but did not pass with vsphere6-nested. Their seems to be some error causing ssh connection to be reset, could not reproduce it locally to understand why is it failing in PRT. And the most recent run 17322 failed because of some error with sprout.

**Pull request merged**